pkg/minikube/cluster/cluster.go

```diff
@@ -342,6 +339,100 @@ func createHost(api libmachine.API, config MachineConfig) (*host.Host, error) {
 	return h, nil
 }
 
+func getDriverOptions(config MachineConfig) (drivers.DriverOptions, error) {
+	switch config.VMDriver {
+	case "virtualbox":
+		d := virtualbox.NewDriver(constants.MachineName, constants.Minipath)
+		machineConfigOptions := map[string]interface{}{
+			"virtualbox-boot2docker-url": config.GetISOFileURI(),
+			"virtualbox-memory":          config.Memory,
+			"virtualbox-cpu-count":       config.CPUs,
+			"virtualbox-disk-size":       config.DiskSize,
+			"virtualbox-hostonly-cidr":   config.HostOnlyCIDR,
+		}
+
+		return createDriverOptions(d, machineConfigOptions)
+	case "vmwarefusion":
+		if minishiftOs.CurrentOS() != minishiftOs.DARWIN {
+			atexit.ExitWithMessage(1, "vmwarefusion driver is only supported on Darwin hosts.")
```

Comment:
I think we should modify the error message as "vmwarefusion is not supported on %s", minishiftOs.CurrentOS(). Then it will be more meaningful